### DISCUSSION TITLE
How to do a simple subtraction

### DISCUSSION THREAD

**Discussion by hanslammerts** [micropython/micropython#14380] [General]
I'm sorry.
Just started out with micropython (for the Py Pico) and one of the simplest things fail.
Please have a look at the following snippet:

`
from machine import Pin

import utime

def heartbeat(currentms):
  setup = False
  print(currentms)

def loop():
  
  loopTimer = utime.ticks_ms
  maxLoopTime = 0
  maxLoopTime = max(maxLoopTime, (utime.ticks_ms - loopTimer))
  loopTimer = utime.ticks_ms

  currentMillis = utime.ticks_ms
  heartbeat(currentMillis)

if \_\_name\_\_ == '\_\_main\_\_':
  loop()
`

The error I get:

Operator "-" not supported for types "() -> int" and "() -> int"

How do I subtract loopTimer from utime.ticks_ms ?
Also, the code backticks don't seem to work....
Indentation disappears....
Thanks.

**Comment by dhalbert** [comment]
Use `utime.ticks_ms()`, with parentheses. You have to call it as a function: it's not a property. Also note now `time` instead of `utime` is preferred.

**Comment by scruss** [comment]
Also, you can't reliably use simple subtraction on ticks counters. These roll over at around 2**30. Use the [time.ticks_diff(ticks1, ticks2)](https://docs.micropython.org/en/v1.22.0/library/time.html#time.ticks_diff) function instead.

Here's what I think your code should look like:
```py
from machine import Pin
import time


def heartbeat(currentms):
    setup = False
    print(currentms)


def loop():
    loopTimer = time.ticks_ms
    maxLoopTime = 0
    maxLoopTime = max(
        maxLoopTime, (time.ticks_diff(time.ticks_ms(), loopTimer))
    )
    loopTimer = time.ticks_ms()
    currentMillis = time.ticks_ms()
    heartbeat(currentMillis)


if __name__ == "__main__":
    loop()
```

(though impossible to tell since indentation was lost. Please read the [Welcome (Please read before posting)](https://github.com/orgs/micropython/discussions/9111) entry on how to post code so it's readable.)
